tools/devtool

```diff
@@ -1163,28 +1163,47 @@ cmd_build_kernel() {
     create_dir "$kernel_dir_host"
 
     # Extract the kernel version from the config file provided as parameter.
-    KERNEL_VERSION=$(grep -Po "^# Linux\/$cfg_pattern (([0-9]+.)[0-9]+)" "$KERNEL_CFG" | cut -d ' ' -f 3)
-    validate_kernel_version "$KERNEL_VERSION"
+    kernel_version=$(grep -Po "^# Linux\/$cfg_pattern (([0-9]+.)[0-9]+)" "$kernel_config" | cut -d ' ' -f 3)
+    validate_kernel_version "$kernel_version"
+
+    kernel_major_version="${kernel_version%%.*}"
+    kernel_url_base="https://cdn.kernel.org/pub/linux/kernel"
+    kernel_latest_version=$( \
+        curl -s $kernel_url_base/v$kernel_major_version.x/ \
+        | grep -o "linux-$kernel_version\.[0-9]*\.tar.xz" \
+        | sort -rn -t . -k 3 \
+	| head -n 1)
+    kernel_url="$kernel_url_base/v$kernel_major_version.x/$kernel_latest_version"
 
-    recipe_commit="b551cccc405a73a6d316c0c09dfe0b3e7a73ba3f"
-    recipe_url="https://raw.githubusercontent.com/rust-vmm/vmm-reference/$recipe_commit/resources/kernel/make_kernel.sh"
+    echo "Downlaoding kernel from $kernel_url"
     run_devctr \
         --user "$(id -u):$(id -g)" \
         --workdir "$kernel_dir_ctr" \
-        -- /bin/bash -c "curl -LO $recipe_url && source make_kernel.sh && extract_kernel_srcs $KERNEL_VERSION"
-    ok_or_die "Could not extract kernel sources with recipe $recipe_url!"
-
-    cp "$KERNEL_CFG" "$kernel_dir_host/linux-$KERNEL_VERSION/.config"
-
-    KERNEL_BINARY_NAME="vmlinux-$KERNEL_VERSION-$arch.bin"
+        -- /bin/bash -c "curl $kernel_url > $kernel_latest_version"
```

Comment:
Maybe we want to keep the `-L`, to honor redirects.

Reply:
Indeed. Thanks!

Reply:
Fixed.